lib/redbird/plug/session/redis.ex

```diff
@@ -13,26 +15,30 @@ defmodule Plug.Session.REDIS do
     opts
   end
 
-  def get(_conn, namespaced_key, _init_options) do
-    case get(namespaced_key) do
-      :undefined -> {nil, %{}}
-      value -> {namespaced_key, value |> :erlang.binary_to_term()}
+  def get(conn, prospective_key, _init_options) do
+    with {:ok, key, _} <- Key.extract_key(prospective_key),
+         {:ok, _verified_key} <- Key.verify(key, conn),
+         value when is_binary(value) <- get(prospective_key) do
+      {prospective_key, Crypto.safe_binary_to_term(value)}
```

Comment:
It seems like this is the only place we use `Redbird.Crypto` in `Session.REDIS`, right? And I saw we use `Redbird.Crypto` inside `Redbird.Key` for other reasons. 

What do you think about pushing this functionality into `Redbird.Key` so that only `Redbird.Key` depends on `Redbird.Crypto`? 

Then the dependencies are like this:

```
Plug.Session.REDIS -> Redbird.Key -> Redbird.Crypto
```

Maybe something like `Redbird.Key.decode/1` would capture what we're doing?

Reply:
I follow you logic around the places in which the `Crypto` module shows up as a dependency. I hesitate to move it to `Key` as I intend that module to encapsulate logic around the Redis key and this particular call deals with the value and not the key. 

I'll give it some thought.

Reply:
That's fair. Yeah, if you like the responsibilities where they are, I think it makes sense not to do what I was thinking. I think my suggestion was a bit mechanical (in the sense that I asked, why is this here and it could not be?), but the boundaries for the modules as you explain them, make sense.